connectors/src/connectors/snowflake/temporal/activities.ts

```diff
@@ -1,7 +1,155 @@
+  if (tablesRes.isErr()) {
+    throw tablesRes.error;
+  }
+
+  const tablesValidation = t.array(snowflakeTableCodec).decode(tablesRes.value);
+  if (isLeft(tablesValidation)) {
+    const pathError = reporter.formatValidationErrors(tablesValidation.left);
+    throw new Error(`Invalid tables response: ${pathError}`);
+  }
+  const [allDatabases, allSchemas, allTables] = await Promise.all([
+    RemoteDatabaseModel.findAll({
+      where: {
+        connectorId,
+      },
+    }),
+    RemoteSchemaModel.findAll({
+      where: {
+        connectorId,
+      },
+    }),
+    RemoteTableModel.findAll({
+      where: {
+        connectorId,
+      },
+    }),
+  ]);
+
+  const readGrantedInternalIds = new Set([
+    ...allDatabases.map((db) => db.internalId),
+    ...allSchemas.map((s) => s.internalId),
+    ...allTables.map((t) => t.internalId),
+  ]);
+  const tableByInternalId = Object.fromEntries(
+    allTables.map((table) => [table.internalId, table])
+  );
+
+  const parseTableInternalId = (
+    tableInternalId: string
+  ): [string, string, string] => {
+    const [dbName, schemaName, tableName] = tableInternalId.split(".");
+    if (!dbName || !schemaName || !tableName) {
+      throw new Error(`Invalid table internalId: ${tableInternalId}`);
+    }
+
+    return [dbName, schemaName, tableName];
+  };
+
+  const isTableReadGranted = (tableInternalId: string) => {
+    const [dbName, schemaName] = parseTableInternalId(tableInternalId);
+
+    const schemaInternalId = [dbName, schemaName].join(".");
+
+    return (
+      readGrantedInternalIds.has(dbName) ||
+      readGrantedInternalIds.has(schemaInternalId) ||
+      readGrantedInternalIds.has(tableInternalId)
+    );
+  };
+
+  const tablesOnSnowflake = tablesValidation.right;
+  const internalIdsOnSnowflake = new Set(
+    tablesOnSnowflake.map(
+      (t) => `${t.database_name}.${t.schema_name}.${t.name}`
+    )
+  );
+
+  for (const internalId of internalIdsOnSnowflake) {
+    if (isTableReadGranted(internalId)) {
+      let table = tableByInternalId[internalId];
+      if (!table || !table.lastUpsertedAt) {
+        if (!table) {
+          const [dbName, schemaName, tableName] =
+            parseTableInternalId(internalId);
+          table = await RemoteTableModel.create({
+            connectorId,
+            internalId,
+            permission: "inherited",
+            name: tableName,
+            schemaName,
+            databaseName: dbName,
+          });
+        }
+
+        await upsertTable({
+          dataSourceConfig: dataSourceConfigFromConnector(connector),
+          tableId: internalId,
+          tableName: internalId,
+          remoteDatabaseTableId: internalId,
+          remoteDatabaseSecretId: connector.connectionId,
+          // TODO(SNOWFLAKE): decide what to do wrt description.
+          tableDescription: "",
+          parents: [
+            `${table.databaseName}.${table.schemaName}`,
+            table.databaseName,
```

Comment:
(does order matter?)

Reply:
I heard that it does

Reply:
But I don't know for sure cc @PopDaph

Reply:
If yes: put a comment.

Reply:
I think so cause of https://github.com/dust-tt/dust/pull/6950